bin/compile

```diff
@@ -1,91 +1,76 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env bash
 
-require 'yaml'
+# Usage:
+#
+#     $ bin/compile <build-dir> <env-path>
 
-$stdout.sync = true
+# Fail fast and fail hard.
+set -eo pipefail
 
-def indent(msg)
-  puts "       #{msg}"
-end
-
-BUILD_DIR = ARGV[0]
-ENV_DIR   = ARGV[2]
-
-# Colours for shell colouring
-YELLOW  = '\033[1;33m'
-RESET   ='\033[0m'
-
-def warning message
-  system "echo '#{YELLOW}WARNING: #{message}#{RESET}'"
-end
-
-puts "-----> Found an .oracle.yml"
-
-ORACLE_CONFIG = YAML.load_file("#{BUILD_DIR}/.oracle.yml")
+BUILD_DIR=$1
+ENV_DIR=$2
+BIN_DIR=$(cd "$(dirname "$0")"; pwd) # absolute path
 
+ORACLE_CONFIG="${BUILD_DIR}/.oracle.yml"
 ORACLE_INSTANT_CLIENT_DIR="vendor/oracle_instantclient"
 ORACLE_INSTANT_CLIENT_URL="http://buildpacks.teresa-apps.io/oracle/oracle-instantclient-full-12.1-linux.x64.tar.gz"
-ORACLE_INSTANT_CLIENT_FILENAME=ORACLE_INSTANT_CLIENT_URL.split(/\//).last
+ORACLE_INSTANT_CLIENT_FILENAME=$(echo $ORACLE_INSTANT_CLIENT_URL | cut -d/ -f5)
 
-indent "Making directory #{BUILD_DIR}/#{ORACLE_INSTANT_CLIENT_DIR}"
-`mkdir -p #{BUILD_DIR}/#{ORACLE_INSTANT_CLIENT_DIR}`
+source "$BIN_DIR/utils"
 
-indent "Downloading and extracting #{ORACLE_INSTANT_CLIENT_FILENAME}"
-result = `curl #{ORACLE_INSTANT_CLIENT_URL} -L -s -o - | tar -xvz -C #{BUILD_DIR}/#{ORACLE_INSTANT_CLIENT_DIR} -f - `
+puts-step "Found an .oracle.yml"
```

Comment:
What about an if here to check if that file exists?

Reply:
(I know, there is the `detect` step, but, I don't know if with multi-buildpacks deploy the detect phase will run)

Reply:
if the `oracle.yml` don't exist the buildpack fail, even with multi-buildpacks

Reply:
LGTM 👍